src/streaming/controllers/BufferController.js

```diff
@@ -306,6 +300,34 @@ function BufferController(config) {
         }
     }
 
+    function adjustSeekTarget () {
+        // Check buffered data only for audio and video
+        if (type !== Constants.AUDIO && type !== Constants.VIDEO) return;
+        if (isNaN(seekTarget)) return;
+
+        const segmentDuration = representationController.getCurrentRepresentation().segmentDuration;
+        const currentTime = playbackController.getTime();
+
+        // Check if current buffered range already contains seek target (and current video element time)
+        let range = getRangeAt(seekTarget, 0);
+        if (currentTime === seekTarget && range) return;
+
+        // Get buffered range corresponding to the seek target
+        range = getRangeAt(seekTarget, segmentDuration);
+        if (!range) return;
+
+        if (Math.abs(currentTime - seekTarget) > segmentDuration) {
+            // If current video model time is decorrelated from seek target (and appended buffer) then seek video element to seek target
+            // (in case of live streams on some browsers/devices for which we can't set video element time at unavalaible range)
+            playbackController.seek(Math.max(seekTarget, range.start), false, true);
```

Comment:
Do we need to make sure here that seekTarget is <= range.end?

Reply:
Yes you're right, appended segment can be anterior from seek target. 
I did the change, please check